2015-Fall/lrusnac/CalculatorV2/CalculatorV2/CalculatorModel.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  CalculatorModel.swift
+//  CalculatorV2
+//
+//  Created by Leonid Rusnac on 08/09/15.
+//  Copyright (c) 2015 Leonid Rusnac. All rights reserved.
+//
+
+import Foundation
+
+class CalculatorModel {
+
+    private enum Op: Printable{
+        case Operand(Double)
+        case UnaryOperator(String, Double -> Double)
+        case BinaryOperator(String, (Double, Double) -> Double)
+        case Constant(String, Double)
+        
+        // equivalent of toString in java
+        var description: String {
+            get {
+                switch self {
+                case .Operand(let operand):
+                    return "\(operand)"
+                case .UnaryOperator(let operatorSymbol, _):
+                    return operatorSymbol
+                case .BinaryOperator(let operatorSymbol, _):
+                    return operatorSymbol
+                case .Constant(let operatorSymbol, _):
+                    return operatorSymbol
+                }
+            }
+        }
+    }
+    
+    private var ops = [Op]()
+    private var knownOps = [String:Op]()
+    
+    
+    init() {
+        func learnOp(op: Op) {
+            knownOps[op.description] = op
+        }
+        
+        learnOp(Op.BinaryOperator("+", +))
+        learnOp(Op.BinaryOperator("×", *))
+        learnOp(Op.BinaryOperator("÷", {$1/$0}))
+        learnOp(Op.BinaryOperator("-", {$1-$0}))
+        
+        learnOp(Op.UnaryOperator("√", sqrt))
+        learnOp(Op.UnaryOperator("sin", sin))
+        learnOp(Op.UnaryOperator("cos", cos))
+        
+        learnOp(Op.Constant("π", M_PI))
+    }
+    
+    func pushNumber(number: Double) {
+        ops.append(Op.Operand(number))
+    }
+    
+    func pushOpNotNumber(keyString: String) {
+        if let con = knownOps[keyString] {
+            ops.append(con)
+        }
+    }
+    
+    func clear() {
+        ops.removeAll(keepCapacity: false)
```

Comment:
You are on the right track. The capacity is how much memory is allocated for the array. As expanding the memory allocated for expandable arrays can be expensive, it sometimes make sense pre-determine how memory is allocated. I'll gladly go in-depth with you on this at some point if you wanna learn the specifics of this. But as you (probably) don't intend on messing around with the memory, you are right to just go with the default value.

Also, if the default value is false, why specify it?